### PR TITLE
Add `@custom-media` support

### PR DESCRIPTION
**Pull request by MoOx**
It's my first contribution to css-parse, I hope it's ok.

Close #82

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -382,6 +382,28 @@ module.exports = function(css, options){
...
+  /**
+  * Parse custom-media.
+  */
+
+  function atcustommedia() {
+    var pos = position();
+    var m = match(/^@([-\w]+)?custom-media (--[^\s]+) *([^{]+);/);
+    if (!m) return;
+
+    var vendor = trim(m[1]);
```
Comment (necolas) [inline index.js]: don't think this is necessary

File: index.js [hunk 2]
```diff
@@ -382,6 +382,28 @@ module.exports = function(css, options){
     });
   }
 
+
+  /**
+  * Parse custom-media.
```
Comment (necolas) [inline index.js]: indent by one more space :)

**Comment by necolas** [comment]
Hmm, I was thinking we needed to do some more significant rethinking of the @-rule parsing, to better represent arbitrary @-rules rather than having to modify the parser each time there are new @-rule additions to CSS. If I want to create my own @-rule, that isn't part of CSS, the parser blows up. (e.g. https://github.com/ai/postcss#atrule-node)

But wouldn't be opposed to adding this as a stop-gap.

**Comment by tj** [comment]
@necolas yea true good point, sounds good to me

File: index.js (hunk 1 above)
Comment (MoOx) [inline index.js]: you mean `var`s decalrations & directly use the result below ?

File: index.js (hunk 2 above)
Comment (MoOx) [inline index.js]: Yep I'll do that sorry.

**Comment by MoOx** [comment]
I agree with the fact that we should handle new (unknown) atrules like postcss. I did that really quickly without thinking.
Maybe we can just keep this new one & add a simple atrules fallback (name + params) so we can keep specific atrules (like keyframes) the way they are today ? With this method we will be able to parse new atrules but at the same times handle specific rules to have a nicer api don't you think ?
Beside that, keyframes could maybe condisdered as a sort of nesting (we will need that soon don't we ?)

**Comment by necolas** [comment]
Any objections to this going in?

**Pull request merged**